docs/source/links/rest.md

```diff
@@ -0,0 +1,394 @@
+
+<h2 id="rest">@rest directive</h2>
+
+This is where you setup the endpoint you want to fetch.
+The rest directive could be used at any depth in a query, but once it is used, nothing nested in it can be GraphQL data, it has to be from the `RestLink` or other resource (like the [`@client` directive](./state.html))
+
+<h3 id="rest.arguments">Arguments</h3>
+
+An `@rest(…)` directive takes two required and several optional arguments:
+
+* `type: string`: The GraphQL type this will return
+* `path: string`: uri-path to the REST API. This could be a path or a full url. If a path, the endpoint given on link creation or from the context is concatenated with it to produce a full `URI`.
+* _optional_ `method?: "GET" | "PUT" | "POST" | "DELETE"`: the HTTP method to send the request via (i.e GET, PUT, POST)
+* _optional_ `endpoint?: string` key to use when looking up the endpoint in the (optional) `endpoints` table if provided to RestLink at creation time.
+* _optional_ `bodyKey?: string = "input"`: This is the name of the `variable` to use when looking to build a REST request-body for a `PUT` or `POST` request. It defaults to `input` if not supplied.
+* _optional_ `bodyBuilder?: /function/`: If provided, this is the name a `function` that you provided to `variables`, that is called when a request-body needs to be built. This lets you combine arguments or encode the body in some format other than JSON.
+
+<h3 id="rest.arguments.variables">Variables</h3>
+
+You can use query `variables` inside nested queries, or in the the path argument of your directive:
+
+```graphql
+query postTitle {
+  post(id: "1") @rest(type: "Post", path: "/post/:id") {
+    id
+    title
+  }
+}
+```
+
+<h4 id="rest.arguments.body">`bodyKey` / `bodyBuilder`</h4>
+
+When making a `POST` or `PUT` HTTP request, you often need to provide a request body. By [convention](https://graphql.org/graphql-js/mutations-and-input-types/), GraphQL recommends you name your input-types as `input`, so by default that's where we'll look to find a JSON object for your body.
+
+<h5 id="rest.arguments.body.key">`bodyKey`</h5>
+
+If you need/want to name it something different, you can pass `bodyKey`, and we'll look at that variable instead.
+
+In this example the publish API accepts a body in the variable `body` instead of input:
+
+```graphql
+mutation publishPost(
+  $someApiWithACustomBodyKey: PublishablePostInput!
+) {
+  publishedPost: publish(input: "Foo", body: $someApiWithACustomBodyKey)
+    @rest(
+      type: "Post"
+      path: "/posts/:input/new"
+      method: "POST"
+      bodyKey: "body"
+    ) {
+    id
+    title
+  }
+}
+```
+
+[Unit Test](https://github.com/apollographql/apollo-link-rest/blob/c9d81ae308e5f61b5ae992061de7abc6cb2f78e0/src/__tests__/restLink.ts#L1803-L1846)
+
+<h5 id="rest.arguments.body.builder">`bodyBuilder`</h5>
```

Comment:
Our stylesheet makes h5's look odd. Let's make this an h4

Reply:
For semantic structural reasons we chatted about leaving this for now, and letting a visual designer fix the stylesheet in the future!